app/services/coin_market_cap_pro/coin_market_cap_pro_helpers.rb

```diff
@@ -11,5 +11,41 @@ def get_error_message(contents)
     def currency
       return "USD"
     end
+
+    def get_default_api_headers
+      @default_api_headers ||= { "X-CMC_PRO_API_KEY" => ENV.fetch('COINMARKETCAP_API_KEY') }
+    end
+
+    def extract_api_data(response, healthcheck_url = nil)
+      contents = JSON.parse(response.body)
+
+      # ping health check if api error
+      json_response_code = get_json_response_code(contents)
+
+      if response.success? && json_response_code == 0 then
+        return contents['data']
+      end
+
+      error_message = "ERROR HTTP(#{response.code}) JSON(#{json_response_code}): #{get_error_message(contents)}"
+      if healthcheck_url.present?
+        Net::HTTP.post(URI.parse("#{healthcheck_url}/fail"), error_message)
+      else
+        puts error_message
```

Comment:
Does it make sense to throw a Rollbar error here in case `healthcheck_url` is not present? Genuinely asking.

Reply:
I made healthchecks optional for helpers in case some things don't use healthchecks.
I think at the service level, we're fetching with `ENV.fetch`, so I assume this would lead to rollbar firing if it was needed but missing?